lib/exchanges.js

```diff
@@ -0,0 +1,94 @@
+var base      = require('taskcluster-base');
+var assert    = require('assert');
+var _         = require('lodash');
+
+// Common schema prefix
+var SCHEMA_PREFIX_CONST = 'http://schemas.taskcluster.net/github/v1/';
+
+/** Build common routing key construct for `exchanges.declare` */
+var commonRoutingKey = function(options) {
+    options = options || {};
+     let routingKey = [
+        {
+          name:             'routingKeyKind',
+          summary:          "Identifier for the routing-key kind. This is " +
+                            "always `'primary'` for the formalized routing key.",
+          constant:         'primary',
+          required:         true
+        }, {
+          name:             'organizationName',
+          summary:          "The GitHub `organizationName` which had an event.",
+          maxSize:          100,
+          required:         true
+        }, {
+          name:             'repositoryName',
+          summary:          "The GitHub `repositoryName` which had an event.",
+          maxSize:          100,
+          required:         true
+        }
+    ]
+    if (options.hasActions) {
+      routingKey.push({
+          name:             'action',
+          summary:          "The GitHub `action` which triggered an event.",
```

Comment:
There are events for assigned, unassigned, labeled, unlabeled, opened, closed, reopened, or synchronized. Listeners can still filter by using primary.#.#.opened right? If that's too cumbersome, I can see the value of only supporting notifications for open and closed. Otherwise, I was hoping to support all the available actions.